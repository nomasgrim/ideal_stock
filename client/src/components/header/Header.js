import { Link } from 'react-router-dom'
import AppBar from '@material-ui/core/AppBar'
import Button from '@material-ui/core/Button'
import React, { Fragment } from 'react'
import Toolbar from '@material-ui/core/Toolbar'

import './header.css'
import logo from '../../logo.png'

/**
 * @function
 * @description component for header
 * @return {ReactElement} JSX
 */
const Header = () => (
  <Fragment>
    <div className='header'>
      <AppBar position='static'>
        <Toolbar>
          <img src={logo} height={40} alt='logo' />
          <Button color='inherit'>
            <Link to='/' style={{ color: 'white' }}>WebSocket Examples</Link>
          </Button>
          <Button color='inherit'>
            <Link to='/examples' style={{ color: 'white' }}>New Route</Link>
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  </Fragment>
)

export default Header
