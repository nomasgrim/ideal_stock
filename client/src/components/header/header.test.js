/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Header from './Header'

test('Renders the header component', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
