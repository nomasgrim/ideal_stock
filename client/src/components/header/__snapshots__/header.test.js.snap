// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the header component 1`] = `
<div
  className="header"
>
  <header
    className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7"
  >
    <div
      className="MuiToolbar-root-36 MuiToolbar-gutters-37"
    >
      <img
        alt="logo"
        height={40}
        src="logo.png"
      />
      <button
        className="MuiButtonBase-root-62 MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-39"
        >
          <a
            href="/"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
          >
            Home
          </a>
        </span>
        <span
          className="MuiTouchRipple-root-65"
        />
      </button>
      <button
        className="MuiButtonBase-root-62 MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-39"
        >
          <a
            href="/examples"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
          >
            WebSocket Examples
          </a>
        </span>
        <span
          className="MuiTouchRipple-root-65"
        />
      </button>
    </div>
  </header>
</div>
`;
